manifests/install.pp

```diff
@@ -33,12 +33,18 @@
     require => Exec["pyenv::checkout ${user}"],
   }
 
-  exec { "pyenv::shrc ${user}":
-    command => "echo 'source ${pyenvrc}' >> ${shrc}",
-    user    => $user,
+  file { $shrc:
```

Comment:
@abloomston Your issue on travis is caused by the old spec since you removed the `Exec[pyenv::shrc ${user}]`.

I agree with @gwax, we should prevent duplicate resource definition error, his workaround looks good, but is known as an antipattern. See [PUP-1968](https://tickets.puppetlabs.com/browse/PUP-1968?focusedCommentId=68416&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-68416).
Or maybe we can use virtual resources to avoid duplicate `File[$shrc]` declarations?
https://docs.puppet.com/puppet/latest/lang_virtual.html